src/gflags.cc

```diff
@@ -800,7 +826,7 @@ CommandLineFlag* FlagRegistry::SplitArgumentLocked(const char* arg,
                                     kError, key->c_str());
       return NULL;
     }
-    if (strcmp(flag->type_name(), "bool") != 0) {
+    if (!flag->OfType<bool>()) {
```

Comment:
Why not just use a getter function which returns the `FlagValue::ValueType` of the `type_` member variable? The test would then be something like:
```
if (flag->Type() != FlagValue::FV_BOOL)
```
It seems the creators of gflags indeed wanted to limit the use of template functions where possible to not increase the `.o` file size (see [this comment](https://github.com/dreamer-dead/gflags/blob/46ea10f071fa6e4bdd6a5d6c789da02ee424f47c/src/gflags.cc#L266)). I think a slightly increased binary file size is worth it in case of the `FlagRegisterer` constructor for the benefits of it, but maybe not for this `OfType` function. So instead of `bool FlagValue::OfType<T>`, I would add a `FlagValueType FlagValue::Type()` function. No need for the template-style syntactic sugar here. It doesn't really make it more readable than the above, but as well makes it much cleaner and safer than a `strcmp`.